smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -70,15 +71,18 @@
     private final ServiceShape service;
     private final FileManifest fileManifest;
     private final SymbolProvider symbolProvider;
-    private final SymbolProvider serverSymbolProvider;
     private final Model nonTraits;
     private final TypeScriptDelegator writers;
     private final List<TypeScriptIntegration> integrations = new ArrayList<>();
     private final List<RuntimeClientPlugin> runtimePlugins = new ArrayList<>();
     private final ProtocolGenerator protocolGenerator;
     private final ApplicationProtocol applicationProtocol;
 
-    CodegenVisitor(PluginContext context) {
+    CodegenVisitor(
+            PluginContext context,
+            BiFunction<Model, TypeScriptSettings, SymbolProvider> createSymbolProvider,
+            TypeScriptSettings.ArtifactType artifactType
```

Comment:
This BiFunction is always static, just make createSymbolProvider a method on the ArtifactType enum and you don't have this ugliness.

Reply:
This what you're thinking?